CSharpSource/Source/Microsoft.Xbox.Services.UWP.CSharp/Social/Manager/SocialManager.cs

```diff
@@ -26,19 +25,14 @@ public void AddLocalUser(XboxLiveUser user, SocialManagerExtraDetailLevel extraD
         {
             if (user == null) throw new ArgumentNullException("user");
 
-            // Allocates memory for returned objects
-            IntPtr cErrMessage = Marshal.AllocHGlobal(Marshal.SizeOf<IntPtr>());
-
+            IntPtr cErrMessage;
             // Invokes the c method
-            XSAPI_RESULT errCode = SocialManagerAddLocalUser(user.Impl.XboxLiveUserPtr, extraDetailLevel, cErrMessage);
+            XSAPI_RESULT errCode = SocialManagerAddLocalUser(user.Impl.XboxLiveUserPtr, extraDetailLevel, out cErrMessage);
```

Comment:
>out cErrMessage [](start = 106, length = 15)

How does this work if you don't allocate memory?

Reply:
Basically its doing the same thing but if you specify out/ref C# handles it for you.  See an example here [https://stackoverflow.com/questions/1422701/how-to-p-invoke-when-pointers-are-involved](https://stackoverflow.com/questions/1422701/how-to-p-invoke-when-pointers-are-involved)

---
In reply to: [157104301](https://github.com/Microsoft/xbox-live-unity-plugin/pull/178#discussion_r157104301) [](ancestors = 157104301)

Reply:
To be clear, the way you had written it also works, this is just a bit simpler to understand and write.

---
In reply to: [157273306](https://github.com/Microsoft/xbox-live-unity-plugin/pull/178#discussion_r157273306) [](ancestors = 157273306,157104301)